sql/core/src/main/scala/org/apache/spark/sql/Column.scala

```diff
@@ -68,6 +68,16 @@ class Column(protected[sql] val expr: Expression) extends Logging {
   override def hashCode: Int = this.expr.hashCode
 
   /**
+   * An expression that gets an item at a position out of an [[ArrayType]],
```

Comment:
Extracts a value or values from a complex type.  The following types of extraction are supported:
 - Given an `Array`, an integer ordinal can be used to retrieve a single value.
 - Given a `Struct`, a string fieldname can be used to extract that field.
 - Given an `Array` of `Struct`s, a string can be used to "map" the array to a new array containing only the specified field from each `Struct`.
 - Given a `Map`, a key of the correct type can be used to retrieve an individual value.